torch_sim/runners.py

```diff
@@ -464,14 +491,34 @@ def static(
     )
 
     @dataclass
-    class StaticState(type(state)):
+    class _StaticState(type(state)):
         energy: torch.Tensor
         forces: torch.Tensor | None
         stress: torch.Tensor | None
 
+    # prop_calculator: dict[int, dict[str, _]], we want dict[str, _]
+    extra_prop_keys = [
+        prop_key
+        for prop_key in next(iter(trajectory_reporter.prop_calculators.values()))
+        if prop_key not in ["energy", "forces", "stress"]
+    ]
+    # dynamically extend class fields for extra properties
+    StaticState = make_dataclass(
+        cls_name="StaticState",
+        fields=[(prop_key, torch.Tensor, None) for prop_key in extra_prop_keys],
+        bases=(_StaticState,),
+    )
```

Comment:
Could name `extra_prop_keys ` -> `prop_keys` and then just add "energy", "forces", and "stress" if they aren't present, that'd get away from the need to define the `_StaticState`.